### PR TITLE
[nextest-runner] add a shutdown grace period, with special handling for no grace

### PR DESCRIPTION
**Pull request by saethlin**
This also skips output of the TERMINATING message when `grace-period = "0s"`:
```
    Starting 6 tests across 2 binaries
     TIMEOUT [   2.012s] unicode-xid::exhaustive_tests all_valid_chars_do_not_panic_for_is_xid_continue

--- STDOUT:              unicode-xid::exhaustive_tests all_valid_chars_do_not_panic_for_is_xid_continue ---

running 1 test
test all_valid_chars_do_not_panic_for_is_xid_continue ... 
     TIMEOUT [   2.012s] unicode-xid::exhaustive_tests all_valid_chars_do_not_panic_for_is_xid_start

--- STDOUT:              unicode-xid::exhaustive_tests all_valid_chars_do_not_panic_for_is_xid_start ---

running 1 test
test all_valid_chars_do_not_panic_for_is_xid_start ... 
        PASS [   0.802s]                   unicode-xid tests::test_is_not_xid_continue
        PASS [   0.782s]                   unicode-xid tests::test_is_not_xid_start
        PASS [   0.779s]                   unicode-xid tests::test_is_xid_continue
        PASS [   0.782s]                   unicode-xid tests::test_is_xid_start
------------
     Summary [   7.170s] 6 tests run: 4 passed, 2 timed out, 0 skipped
     TIMEOUT [   2.012s] unicode-xid::exhaustive_tests all_valid_chars_do_not_panic_for_is_xid_continue
     TIMEOUT [   2.012s] unicode-xid::exhaustive_tests all_valid_chars_do_not_panic_for_is_xid_start
error: test run failed
```
I would prefer if this printed exactly `2.000s` instead of what looks like one tokio scheduler tick later.

I'm really not sure of these names, happy to adjust things based on what you think.

### CODE REVIEW

File: nextest-runner/src/runner.rs [hunk 1]
```diff
@@ -670,24 +670,28 @@ impl<'a> TestRunnerInner<'a> {
...
 
-                        let _ = run_sender.send(InternalTestEvent::Slow {
-                            test_instance: test,
-                            retry_data,
-                            // Pass in the slow timeout period times timeout_hit, since stopwatch.elapsed() tends to be
-                            // slightly longer.
-                            elapsed: timeout_hit * slow_timeout.period,
-                            will_terminate,
-                        });
+                        if slow_timeout.grace_period != Duration::ZERO {
```
Comment (sunshowers) [inline nextest-runner/src/runner.rs:673]:
```suggestion
                        if !slow_timeout.grace_period.is_zero()
```

**Review by sunshowers — COMMENTED** [review]
I think this looks really good. I like the `grace-period` name. Just one little comment.

The timeout being printed out exactly would be cool, but let's tackle that in a separate PR.

File: nextest-runner/src/runner.rs (hunk 1 above)
Comment (saethlin) [reply nextest-runner/src/runner.rs]: Ah! This is nice. I adjusted this in a few places.

**Review by sunshowers — APPROVED** [review]
Looks great! Will land later today assuming tests pass. Thanks.

**Pull request merged**